src/components/block/index.ts

```diff
@@ -578,6 +578,29 @@ export default class Block {
       });
   }
 
+  /**
+   * Extracts HTML from Block
+   *
+   * @returns {Promise<string>} - HTML
+   */
+  public async saveAsHTML(): Promise<string> {
+    let extractedBlock = '';
+
+    try {
+      if (this.toolInstance.saveAsHTML) {
+        extractedBlock = await this.toolInstance.saveAsHTML(this.pluginsContent as HTMLElement);
+      } else {
+        extractedBlock = this.holder.innerHTML;
+      }
+    } catch (error) {
+      _.log(`Saving as HTML proccess for ${this.name} tool failed due to the ${error}`, 'log', 'red');
+
+      throw error;
```

Comment:
Rethrow the error.
See: https://wiki.sei.cmu.edu/confluence/display/java/ERR00-J.+Do+not+suppress+or+ignore+checked+exceptions